src/DynamoRevit/DynamoRevit.cs

```diff
@@ -1019,54 +1018,6 @@ public static string GetRevitContext(DynamoRevitCommandData commandData)
 
         #endregion
 
-        #region Exception
-
-        /// <summary>
-        ///     A method to deal with unhandled exceptions.  Executes right before Revit crashes.
-        ///     Dynamo is still valid at this time, but further work may cause corruption.  Here,
-        ///     we run the ExitCommand, allowing the user to save all of their work.
-        /// </summary>
-        /// <param name="sender"></param>
-        /// <param name="args">Info about the exception</param>
-        private static void Dispatcher_UnhandledException(
-            object sender, DispatcherUnhandledExceptionEventArgs args)
-        {
-            args.Handled = true;
-
-            // only handle a single crash per Dynamo sesh, this should be reset in the initial command
-            if (handledCrash)
-                return;
-
-            handledCrash = true;
-
-            string exceptionMessage = args.Exception.Message;
-
-            try
-            {
-                Dynamo.Logging.Analytics.TrackException(args.Exception, true);
-
-                RevitDynamoModel.Logger.LogError("Dynamo Unhandled Exception");
-                RevitDynamoModel.Logger.LogError(exceptionMessage);
-            }
-            catch { }
-
-            try
-            {
-                DynamoModel.IsCrashing = true;
-                RevitDynamoModel.OnRequestsCrashPrompt(
-                    RevitDynamoModel,
-                    new CrashPromptArgs(args.Exception.Message + "\n\n" + args.Exception.StackTrace));
-                RevitDynamoViewModel.Exit(false); // don't allow cancellation
-            }
-            catch { }
-            finally
-            {
-                args.Handled = true;
-                DynamoRevitApp.DynamoButtonEnabled = true;
```

Comment:
what about the button ?

Reply:
The button is reenabled even without this.

Reply:
Is it somewhere in the view.CLose events ?
THere would be a chance that if an exception is thrown in the exception handler, then the button close will not be called at all

Reply:
Yes, it is re-enabled in `OnDynamoViewClosed` [here](https://github.com/DynamoDS/DynamoRevit/blob/fb7c649f0a130869b596bf4ab88dc1cc17c52eb1/src/DynamoRevit/DynamoRevit.cs#L1085).

Reply:
But why the API breaking change so late, in a minor build (3.0.2 vs 3.0.0), and why after Revit API freeze?

This would require more testing on Revit side, that various CER scenarios still work and we are not missing crash reports, possibly even more code changed.

Is this API change in 3.0.2 mandatory or can we revert it, or make other builds based on 3.0.0 with only the critical fixes?

Reply:
Yes, that change is what I was actually referring to. I merged it together with Dynamo 3.0.2.